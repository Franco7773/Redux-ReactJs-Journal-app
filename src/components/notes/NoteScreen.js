import React, { useEffect, useRef } from 'react'
import { NotesAppBar } from './NotesAppBar'
import { useSelector, useDispatch } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { activeNote, startDeleting } from '../../actions/notes';

export const NoteScreen = () => {

  const dispatch = useDispatch();
  const { active: note } = useSelector( state => state.notes );
  const [{ id, body, title, date, url }, handleInputChange, reset ] = useForm( note );
  const activeId = useRef( note.id );

  useEffect(() => {

    if (note.id !== activeId.current) {
      reset( note );
      activeId.current = note.id
    }
  }, [ reset, note ]);

  useEffect(() => {
  
    dispatch( activeNote( id, { id, title, body, date, url }));
  
  }, [ dispatch, id, title, body, date, url ]);
  
  const handleDelete = () => {
    dispatch( startDeleting( id ));
  };

  return (
    <div className="notes__main-content">
      <NotesAppBar />

      <div className="notes__content">
        <input value={ title } onChange={ handleInputChange } name="title" type="text" className="notes__content--title-input" autoComplete="off" placeholder="Some awesome title"></input>

        <textarea value={ body } onChange={ handleInputChange } name="body" className="notes__content--textarea" placeholder="What happened today?"></textarea>

        { url &&
          <div className="notes__content--image">
            <img alt="imagen" src={ url } />
          </div>
        }
      </div>

      <button onClick={ handleDelete } className="btn btn--danger">Delete</button>
      
    </div>
  );
}
