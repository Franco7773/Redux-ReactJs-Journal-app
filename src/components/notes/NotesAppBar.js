import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

  const dispatch = useDispatch();
  const { active } = useSelector( state => state.notes );

  const handleSave = () => {
    dispatch( startSaveNote( active ));
  };

  const handlePictureClick = () => {
    // dispatch( startSaveNote( active ));
    document.querySelector('#fileSelector').click();
  };

  const handleFileChange = ( e ) => {
    e.preventDefault();

    const file = e.target.files[0];

    if (file) {
      dispatch( startUploading( file ));
    }
  };
  
  return (
    <div className="notes__appbar">
      <span>27 de Agosto 2020</span>

      <input id="fileSelector" onChange={ handleFileChange } type="file" name="file" style={{ display: 'none' }}/>

      <div>
        <button onClick={ handlePictureClick } className="btn">Picture</button>
        <button onClick={ handleSave } className="btn">Save</button>
      </div>
    </div>
  );
}
