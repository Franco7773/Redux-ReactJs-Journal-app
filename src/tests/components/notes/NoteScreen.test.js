import React from 'react';
import { Provider } from 'react-redux';

import '@testing-library/jest-dom'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';

import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';


jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn()
}));

const middlewares = [thunk];

const mockStore = configureStore( middlewares );

const initState = {
  auth: {
    uid: '1',
    name: 'Gianfranco'
  },
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    active: {
      id: 1234,
      title: 'Hello',
      body: 'Galaxy',
      date: 0,
      url: 'https://google.com/photo.jpg'
    },
    notes: []
  }
};

const store = mockStore( initState );
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={ store }>
    <NoteScreen />
  </Provider>
);

describe('Pruebas en', () => {
   
  test('Debe de mostrarse correctamente', () => {
    
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('Debe de disparar el activeNote', () => {
    
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'Goodbye'
      }
    });

    expect( activeNote ).toHaveBeenLastCalledWith(
      1234,
      {
        id: 1234,
        title: 'Goodbye',
        body: 'Galaxy',
        date: 0,
        url: 'https://google.com/photo.jpg'
      }
    );
  });
});
