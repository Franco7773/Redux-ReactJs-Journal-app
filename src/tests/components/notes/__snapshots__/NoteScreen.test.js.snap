// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      27 de Agosto 2020
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__content--title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hello"
    />
    <textarea
      className="notes__content--textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value="Galaxy"
    />
    <div
      className="notes__content--image"
    >
      <img
        alt="imagen"
        src="https://google.com/photo.jpg"
      />
    </div>
  </div>
  <button
    className="btn btn--danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
